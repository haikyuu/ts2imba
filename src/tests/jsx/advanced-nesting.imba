import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code2 = '''
function App(){
	return (
		<div>
			<section>
				<div>
					<div>
						<h2>Title</h2>
					</div>
				</div>
			</section>
			<footer>F</footer>
		</div>
	)
}
'''

let imba-code2 = '''
tag App
	def render
		<self>
			<div>
				<section>
					<div>
						<div>
							<h2>
								"Title"
				<footer>
					"F"

'''
test 'multiple levels closed at once' do
	const result = await build tsx-code2
	expect(result.code).toEqual(imba-code2)
	const out = imbac.compile(result.code, sourceId: 'sth')

let tsx-code3 = '''
function App(){
	return (
		<div>
			<div>
				<h3>Title</h3>
			</div>
			<footer>F</footer>
		</div>
	)
}
'''

let imba-code3 = '''
tag App
	def render
		<self>
			<div>
				<div>
					<h3>
						"Title"
				<footer>
					"F"

'''
test 'self closing nesting' do
	const result = await build tsx-code3
	expect(result.code).toEqual(imba-code3)
	const out = imbac.compile(result.code, sourceId: 'sth')



let tsx-code1 = '''
function App(){
	return (
		<div>
			<a href="" />
			<span>Test 2</span>
		</div>
	)
}
'''

let imba-code1 = '''
tag App
	def render
		<self>
			<div>
				<a href="">
				<span>
					"Test 2"

'''
test 'self closing nesting' do
	const result = await build tsx-code1
	expect(result.code).toEqual(imba-code1)
	const out = imbac.compile(result.code, sourceId: 'sth')


let tsx-code = '''
function App(){
	return (
		<div>
			<span>Test</span>
			<div>
				<h2>Title</h2>
				<div>
					<span>
						<a href="" />
						<span>Test 2</span>
						Text
					</span>
				</div>
			</div>
			<footer>F</footer>
		</div>
	)
}
'''

let imba-code = '''
tag App
	def render
		<self>
			<div>
				<span>
					"Test"
				<div>
					<h2>
						"Title"
					<div>
						<span>
							<a href="">
							<span>
								"Test 2"
							"Text"
				<footer>
					"F"

'''
test 'advanced nesting' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')