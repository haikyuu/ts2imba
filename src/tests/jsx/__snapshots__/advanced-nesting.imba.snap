// Vitest Snapshot v1

exports[`advanced nesting 333 1`] = `
"tag App
	<self>
		<div>
			<span>
				\\"Test\\"
			<div>
				<h2>
					\\"Title\\"
				<div>
					<span>
						<a href=\\"\\">
						<span>
							\\"Test 2\\"
						\\"Text\\"
			<footer>
				\\"F\\"
"
`;

exports[`multiple levels closed at once 31 1`] = `
"tag App
	<self>
		<div>
			<section>
				<div>
					<div>
						<h2>
							\\"Title\\"
			<footer>
				\\"F\\"
"
`;

exports[`self closing nesting 1`] = `
"tag App
	<self>
		<div>
			<a href=\\"\\">
			<span>
				\\"Test 2\\"
"
`;

exports[`self closing nesting 12 1`] = `
"tag App
	<self>
		<div>
			<div>
				<h3>
					\\"Title\\"
			<footer>
				\\"F\\"
"
`;
